### ISSUE TITLE
When receiving infinity complex tensor, tf.math.sigmoid outputs NaN

### ISSUE THREAD

**Issue by QuantumCoder4** [tensorflow/tensorflow#63051]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.15.0

### Custom code

Yes

### Current behavior?

Hi, 
I was using tf.math.sigmoid and I notice that this function raises NaN when the input is inf+0.j. 
Such result looks incorrect to me since I believe sigmoid's output should be within the range [0, 1].

### Standalone code to reproduce the issue

```shell
import tensorflow as tf


x = tf.constant([np.inf+0.j], dtype='complex64')
print(tf.keras.activations.sigmoid(x))  # tf.Tensor([nan+nanj], shape=(1,), dtype=complex64)
```

**Comment by QuantumCoder4** [comment]
Thank you for your explanation.
